cli/cli.py

```diff
@@ -0,0 +1,155 @@
+"""
+CLI for interacting with the REMOTE analysis-runner
+    Motivation: https://github.com/populationgenomics/analysis-runner/issues/8
+
+Mostly for interacting with the server/main.py
+
+Called with:
+
+    analysis-runner run \
+        --output-dir gs://some-path/ \
+        --commit-hash <current-hash> \
+        --repo <current-repo>  \
+        --script <default: main.py>
+Process:
+
+    1. Fill in the mising info (repo, commit hash, etc)
+    2. Get the gcloud auth token $(gcloud auth print-identity-token)
+    3. Get the submit URL from the batch parameter
+    4. Form a POST request with params:
+        * output
+        * repo
+        * commit
+        * script
+        * description
+    5. Collect and print response
+"""
+from getpass import getuser
+
+import subprocess
+import click
+import requests
+import google.auth
+import google.auth.transport.requests
+
+BRANCH = 'add-cli'  # 'master'
+DEFAULT_STACK_LOOKUP = (
+    f'https://raw.githubusercontent.com/'
+    f'populationgenomics/analysis-runner/{BRANCH}/cli/stacklookup.json'
+)
+
+
+@click.command()
+@click.option(
+    '--stack',
+    required=True,
+    help='The stack name, which determines which analysis-runner server to send to',
+)
+@click.option(
+    '--output-dir',
+    required=True,
+    help='The output directory of the run, MUST start with gs://',
+)
+@click.option(
+    '--repository',
+    help='The URI of the repository to run, must be approved by the appropriate server.'
+    ' Default behaviour is to find the repository of the current working'
+    ' directory with `git remote get-url origin`',
+)
+@click.option(
+    '--commit-ref',
+    '--hash',
+    help='The hash of repository to run, default behaviour is to '
+    'find the latest hash of the current repository',
```

Comment:
```suggestion
    help='The hash of commit to run, default behaviour is to '
    'use the current commit',
```

I think it would be confusing to use HEAD here instead of whatever the user has checked out currently.